### PR TITLE
played spam

### PR DESCRIPTION
**Pull request by 1010101110**

### CODE REVIEW

File: Hardcore.lua [hunk 1]
```diff
@@ -1336,6 +1341,24 @@ function Hardcore:InitiatePulseCheck()
 	end)
 end
 
+function Hardcore:InitiatePulsePlayed()
+	--init time played
+	Hardcore:RequestTimePlayed()
+
+	--time accumulator
+	C_Timer.NewTicker(TIME_TRACK_PULSE, function()
+		Hardcore_Character.time_tracked = Hardcore_Character.time_tracked + TIME_TRACK_PULSE
```
Comment (aaronma37) [inline Hardcore.lua:1350]: Having this here as opposed to in the requestplayed timer will offset the accumulation by just a bit; but thats probably okay. As long as time played pulse doesn't approach our time gap limit

**Review by aaronma37 — APPROVED** [review]

**Comment by 1010101110** [comment]
yeah i think the only place it matters is here

https://github.com/Zarant/WoW_Hardcore/blob/299f787d171f579a11b0bc705d17ab34f51b3d6e/Hardcore.lua#L611

**Comment by aaronma37** [comment]
> yeah i think the only place it matters is around here
> 
> https://github.com/Zarant/WoW_Hardcore/blob/299f787d171f579a11b0bc705d17ab34f51b3d6e/Hardcore.lua#L611
> 
> in our verification testing was time played vs addon time much different ever?

Generally the gap per session is about 2 to 3 seconds, probably depending on machine and happens during reload.  Over time the difference between addon and playtime grows though

**Comment by RobertB4** [comment]
The problem with this approach is that the difference between time_played and time_tracked can grow by up to `TIME_PLAYED_PULSE` seconds whenever a user logs our or reloads their UI. Worst case, if they log out 59 seconds after the last pulse, the /played kept ticking on the server but time_tracked will not have updated.
Not sure if we should allow for such large gaps, as it can add up over time.

**Comment by 1010101110** [comment]
But we allow 2 minutes every 2 hours so it shouldn't ever get too close?

It was set to 90% originally which is 8 minutes per hour which is not acceptable lol

**Comment by aaronma37** [comment]
> The problem with this approach is that the difference between time_played and time_tracked can grow by up to `TIME_PLAYED_PULSE` seconds whenever a user logs our or reloads their UI. Worst case, if they log out 59 seconds after the last pulse, the /played kept ticking on the server but time_tracked will not have updated. Not sure if we should allow for such large gaps, as it can add up over time.

I'm not sure if I understood, but this PR shouldn't change the overall time_track / time_played right? Just between session gaps

**Comment by 1010101110** [comment]
Could change the played pulse to 30 secs. That means it would have to miss 3 pulses to trigger the alert?

**Comment by aaronma37** [comment]
> Could change the played pulse to 30 secs. That means it would have to miss 3 pulses in a row to trigger the alert?

Did you mean the 10min session gap alert? I think either 30 or 60s is fine. I would be worried if we lower that limit to like 2min

**Comment by RobertB4** [comment]
> I'm not sure if I understood, but this PR shouldn't change the overall time_track / time_played right? Just between session gaps

You are right. I commented at 3am, guess I was half asleep haha. When I got up today my first thought was "couldn't we split the timers?", and after having another look at the PR, this is exactly what it does 👍

**Review by RobertB4 — APPROVED** [review]

**Review by blenderer — COMMENTED** [review]
Just making sure I understand here whats going on. Correct me if I'm wrong:
The main fix here is only recording `/played` every 60 seconds instead of 1 second?

**Comment by RobertB4** [comment]
> Just making sure I understand here whats going on. Correct me if I'm wrong:
The main fix here is only recording /played every 60 seconds instead of 1 second?

Yes, that is more or less correct.
The time_tracked is still updated every second, but the time_played (which is updated using the /played from blizz servers) only gets updated every minute.

@1010101110 correct me if I am wrong!

**Comment by aaronma37** [comment]
> Just making sure I understand here whats going on. Correct me if I'm wrong: The main fix here is only recording `/played` every 60 seconds instead of 1 second?

Yeah,  I think that is the only functional difference

**Comment by aaronma37** [comment]
pulled and tested;

image
after 10 reloads

as expected time gap gets up to 60 extra seconds (which is fine since limit is 10 minutes).  time_played and time_tracked do not diverge more than current implementation

image
after forced (alt+f4) exit. Unfortunate, but we should address this in the future

**Review by blenderer — APPROVED** [review]

**Pull request merged**